--- src/java/davmail/http/DavGatewayHttpClientFacade.java.orig	2011-01-26 15:19:03.000000000 +0100
+++ src/java/davmail/http/DavGatewayHttpClientFacade.java	2011-01-26 16:39:07.000000000 +0100
@@ -57,7 +57,9 @@ import java.util.List;
 public final class DavGatewayHttpClientFacade {
     static final Logger LOGGER = Logger.getLogger("davmail.http.DavGatewayHttpClientFacade");
 
-    static final String IE_USER_AGENT = "Mozilla/4.0 (compatible; MSIE 8.0; Windows NT 6.1; Trident/4.0)";
+    //static final String IE_USER_AGENT = "Mozilla/4.0 (compatible; MSIE 8.0; Windows NT 6.1; Trident/4.0)";
+    //static final String IE_USER_AGENT = "Mozilla/5.0 (X11; U; Linux i686; fr; rv:1.9.2.13) Gecko/20110104 Firefox/3.6.13";
+    static final String IE_USER_AGENT = "Mozilla/5.0 (Windows; U; Windows NT 6.1; fr; rv:1.9.2) Gecko/20100115 Firefox/3.6";
     static final int MAX_REDIRECTS = 10;
     static final Object LOCK = new Object();
     private static boolean needNTLM;
