--- ./apt-pkg/init.cc.orig	2011-07-28 12:39:03.000000000 +0200
+++ ./apt-pkg/init.cc	2011-07-28 12:39:17.000000000 +0200
@@ -76,8 +76,8 @@ bool pkgInitConfig(Configuration &Cnf)
    Cnf.Set("Dir::Media::MountPath","/media/apt");
 
    // State   
-   Cnf.Set("Dir::Log","var/log/apt");
-   Cnf.Set("Dir::Log::Terminal","term.log");
+   Cnf.Set("Dir::Log","var/log");
+   Cnf.Set("Dir::Log::Terminal","apt_dpkg.log");
    Cnf.Set("Dir::Log::History","history.log");
 
    Cnf.Set("Dir::Ignore-Files-Silently::", "~$");
--- ./apt-pkg/deb/dpkgpm.cc.orig	2011-07-28 12:40:55.000000000 +0200
+++ ./apt-pkg/deb/dpkgpm.cc	2011-07-28 12:58:23.000000000 +0200
@@ -662,6 +662,11 @@ bool pkgDPkgPM::OpenLog()
    // open terminal log
    string const logfile_name = flCombine(logdir,
 				   _config->Find("Dir::Log::Terminal"));
+   struct passwd *pw;
+   struct group *gr;
+   pw = getpwnam("root");
+   gr = getgrnam("syslog");
+
    if (!logfile_name.empty())
    {
       term_out = fopen(logfile_name.c_str(),"a");
@@ -669,10 +674,6 @@ bool pkgDPkgPM::OpenLog()
 	 return _error->WarningE("OpenLog", _("Could not open file '%s'"), logfile_name.c_str());
       setvbuf(term_out, NULL, _IONBF, 0);
       SetCloseExec(fileno(term_out), true);
-      struct passwd *pw;
-      struct group *gr;
-      pw = getpwnam("root");
-      gr = getgrnam("adm");
       if (pw != NULL && gr != NULL)
 	  chown(logfile_name.c_str(), pw->pw_uid, gr->gr_gid);
       chmod(logfile_name.c_str(), 0644);
@@ -687,6 +688,8 @@ bool pkgDPkgPM::OpenLog()
       history_out = fopen(history_name.c_str(),"a");
       if (history_out == NULL)
 	 return _error->WarningE("OpenLog", _("Could not open file '%s'"), history_name.c_str());
+      if (pw != NULL && gr != NULL)
+	  chown(history_name.c_str(), pw->pw_uid, gr->gr_gid);
       chmod(history_name.c_str(), 0644);
       fprintf(history_out, "\nStart-Date: %s\n", timestr);
       string remove, purge, install, reinstall, upgrade, downgrade;
